runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/IsmReaderImpl.java

```diff
@@ -491,24 +502,27 @@ public int hashCode() {
 
     inChannel = initializeBloomFilterAndIndexPerShard(inChannel);
```

Comment:
I believe the swap to computeIfAbsent will resolve beam-6559 and allow us to remove SuppressFBWarnings and associated comment.

Reply:
I deleted @SuppressFBWarnings("IS2_INCONSISTENT_SYNC"). I'll monitor the pre-merge checks.

Reply:
The warning still happens with `footer` and `shardOffsetToShardMap` [1]. I'll keep `SuppressFBWarnings("IS2_INCONSISTENT_SYNC")`

[1] https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/22067/spotbugs/